voice/voice-ai/x/SpeechRecognition/OpenAITextToSpeech.swift

```diff
@@ -0,0 +1,63 @@
+
+import Foundation
+import Sentry
+
+class OpenAITextToSpeech: NSObject {
+    
+    var currentDataTask: URLSessionDataTask?
+    
+    func fetchAudioData(text: String, completion: @escaping (Result<Data, Error>) -> Void) {
+        guard let url = URL(string: "https://api.openai.com/v1/audio/speech") else {
+            completion(.failure(NSError(domain: "OpenAITextToSpeechError", code: -1, userInfo: [NSLocalizedDescriptionKey: "Invalid URL"])))
+            return
+        }
+        
+        guard let apiKey = AppConfig.shared.getTextToSpeechKey() else {
+            SentrySDK.capture(message: "OpenAI API key is nil")
+            return
+        }
+        
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        request.addValue("Bearer \(apiKey)", forHTTPHeaderField: "Authorization")
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        
+        let body: [String: Any] = [
+            "model": "tts-1",
+            "input": text,
+            "voice": "nova"
+        ]
+        
+        do {
+            request.httpBody = try JSONSerialization.data(withJSONObject: body)
+        } catch {
+            completion(.failure(error))
+            return
+        }
+        
+        currentDataTask = URLSession.shared.dataTask(with: request) { data, response, error in
```

Comment:
You can stream the response in chunks and play in real time without waiting for the whole file https://platform.openai.com/docs/guides/text-to-speech/streaming-real-time-audio